log_forwarded_for_ip/__init__.py

```diff
@@ -3,13 +3,24 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
 from werkzeug.serving import WSGIRequestHandler
+from mimetools import Message
+from openerp import models
+from .hooks import restore_address_string
 
 
 def address_string(self):
-    forwarded_for = self.headers.get('X-Forwarded-For', '').split(',')
-    if forwarded_for and forwarded_for[0]:
-        return forwarded_for[0]
-    else:
-        return self.client_address[0]
+    if self.headers and isinstance(self.headers, Message):
+        forwarded_for = self.headers.get('X-Forwarded-For', '').split(',')
+        if forwarded_for and forwarded_for[0]:
+            return forwarded_for[0]
+    return self.client_address[0]
 
-WSGIRequestHandler.address_string = address_string
+
+class log_forwarded_for_ip_installed(models.AbstractModel):
+    _name = 'log.forwarded.for.ip.installed'
+
+    def _register_hook(self, cr):
+        if not hasattr(WSGIRequestHandler, '_odoo_str_replaced'):
+            WSGIRequestHandler._odoo_str_replaced = True
```

Comment:
I use something like `_${orignal_name}_org` for my markers, this makes it more clear what it's about if you ever come across the memer in a debugger